collect_app/src/main/java/org/odk/collect/android/dao/helpers/ContentResolverHelper.java

```diff
@@ -78,15 +80,21 @@ public static String getFormPath(Uri uri) {
      * @param fileUri Whose name we want to get
      * @return The file's extension
      */
-    public static String getFileExtensionFromUri(Uri fileUri) {
+    public static String getFileExtensionFromUri(Context context, Uri fileUri) {
         try (Cursor returnCursor = getContentResolver().query(fileUri, null, null, null, null)) {
             if (returnCursor != null && returnCursor.getCount() > 0) {
+                String filename = null;
                 int nameIndex = returnCursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
-                returnCursor.moveToFirst();
-                String filename = returnCursor.getString(nameIndex);
-                // If the file's name contains extension , we cut it down for latter use (copy a new file).
-                if (filename.lastIndexOf('.') != -1) {
+                if (nameIndex != -1) {
```

Comment:
Added just in case because maybe in the future we run into a problem where OpenableColumns.DISPLAY_NAME doesn't exist at all like it was in https://github.com/opendatakit/collect/pull/2613